### PR TITLE
Fbnet retiarii

### PR DESCRIPTION
**Pull request by AnkitRai-22**
The retiarii implementation of fbnet has been done. Few features are yet to be implemented.

### CODE REVIEW

**Comment by AnkitRai-22** [comment]
PR to issue number #3814

File: nni/retiarii/oneshot/pytorch/fbnet.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
+import copy
```
Comment (ultmaster) [inline nni/retiarii/oneshot/pytorch/fbnet.py:1]: Suggest adding a runnable example.

**Comment by QuanluZhang** [comment]
@AnkitRai-22 , thanks for your contribution, this pr looks great, could you check @ultmaster 's comments to see how to make this pr better.

**Comment by AnkitRai-22** [comment]
Thank you, @ultmaster and @QuanluZhang  for your comments. I have a tight schedule first half of this week. I'll surely look into all the suggestions by Friday.

File: nni/retiarii/oneshot/pytorch/fbnet.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+                                                        sampler=valid_sampler,
+                                                        num_workers=self.workers)
+
+    def _train_one_epoch(self, epoch):
+        self.model.train()
+        meters = AverageMeterGroup()
+        for step, ((trn_X, trn_y), (val_X, val_y)) in enumerate(zip(self.train_loader, self.valid_loader)):
+            trn_X, trn_y = to_device(trn_X, self.device), to_device(trn_y, self.device)
+            val_X, val_y = to_device(val_X, self.device), to_device(val_y, self.device)
+
```
Comment (pkubik) [inline nni/retiarii/oneshot/pytorch/fbnet.py:162]: Any reason why the order of phases is swapped in relation to the FBNet paper (Appendix)?
> In each epoch, we first train the operator parameters `w_a` on 80% of the training set (...) Next, we train
the architecture distribution parameter θ on the rest 20% of the training set.

Comment (ultmaster) [reply nni/retiarii/oneshot/pytorch/fbnet.py:162]: Is this a question about FBNet? Maybe you should go to reddit/twitter for discussion?

Comment (pkubik) [reply nni/retiarii/oneshot/pytorch/fbnet.py:162]: Ok, sorry. I thought there was a mistake within this implementation, but I didn't parse the loop correctly.

**Comment by ultmaster** [comment]
We have a dedicated new plan to support FBNet to Retiarii now. Closing this. If you have any more inquiries, feel free to open new issues.

**Pull request closed without merging**